### PR TITLE
[Do not merge] Follow croissant 1.0 specs

### PR DESCRIPTION
**Pull request by ccl-core**

### CODE REVIEW

File: services/api/src/api/routes/croissant.py [hunk 1]
```diff
@@ -122,38 +126,47 @@ def get_croissant_from_dataset_infos(
...
-                        "source": {"distribution": distribution_name, "extract": {"column": column}},
-                    }
+                    _remove_none_values(
+                        {
+                            "@type": "ml:Field",
+                            "@id": _escape_name(column, fields_names) if croissant_1_0 else None,
+                            "name": _escape_name(column, fields_names),
+                            "description": f"Column '{column}' from the Hugging Face parquet file.",
+                            "dataType": HF_TO_CROISSANT_VALUE_TYPE[feature.dtype],
+                            "source": {"fileSet": {"@id": distribution_name}, "extract": {"column": column}},
```
Comment (marcenacp) [inline services/api/src/api/routes/croissant.py:136]: Should you distinguish v0.8 and v1.0 using the flag?

```
"source": {"distribution": distribution_name, "extract": {"column": column}},

"source": {"fileSet": {"@id": distribution_name}, "extract": {"column": column}},
```

File: services/api/src/api/routes/croissant.py [hunk 2]
```diff
@@ -169,50 +182,65 @@ def get_croissant_from_dataset_infos(
...
                 "@vocab": "https://schema.org/",
-                "column": "ml:column",
+                "citeAs": "cr:citeAs",
+                "column": "cr:column",
+                "conformsTo": "dct:conformsTo",
+                "cr": "http://mlcommons.org/croissant/",
                 "data": {
-                    "@id": "ml:data",
-                    "@type": "@json",
+                    "@id": "cr:data",
```
Comment (marcenacp) [inline services/api/src/api/routes/croissant.py:205]: `ml` for 0.8 and `cr` for 1.0?

File: services/api/src/api/routes/croissant.py [hunk 3]
```diff
@@ -231,6 +259,14 @@ def _get_full_jsonld_parameter(request: Request) -> bool:
     return True
 
 
+def _get_use_croissant_1_0(request: Request) -> bool:
+    """Whether the output follows Croissant 1.0 or 0.8. Defaults to True."""
+    use_croissant_1_0 = get_request_parameter(request, "use_croissant_1_0", default="true")
```
Comment (marcenacp) [inline services/api/src/api/routes/croissant.py:264]: Not sure about the snake_case for the URL parameter. From their documentation it looks like HF favors short words. Maybe `?v1=true` or `?isV1=true`?

File: services/api/tests/routes/test_croissant.py [hunk 4]
```diff
@@ -52,16 +52,29 @@
...
+    if use_croissant_1_0:
+        assert "conformsTo" in croissant
     assert croissant["name"] == "user_squad_with_space"
     assert "distribution" in croissant
+    if use_croissant_1_0:
+        for distribution in croissant["distribution"]:
+            assert "@id" in distribution
     assert "recordSet" in croissant
+    if use_croissant_1_0:
+        for record_set in croissant["recordSet"]:
```
Comment (marcenacp) [inline services/api/tests/routes/test_croissant.py:76]: For all for loops, add an assert testing that the list is not empty:

```
assert croissant["distribution"]
```

Otherwise the list can be empty and the test stil pass

File: services/api/src/api/routes/croissant.py (hunk 3 above)
Comment (marcenacp) [reply services/api/src/api/routes/croissant.py:264]: @severo The idea would be to have a boolean flag to first control that the migration to 1.0 is ok (e.g., using [the crawler](https://github.com/mlcommons/croissant/blob/main/health/crawler/spiders/huggingface.py)) and then do the switch. WDYT?

File: services/api/src/api/routes/croissant.py (hunk 2 above)
Comment (ccl-core) [reply services/api/src/api/routes/croissant.py]: Done.

File: services/api/src/api/routes/croissant.py (hunk 3 above)
Comment (ccl-core) [reply services/api/src/api/routes/croissant.py]: Changed to `isV1`, done.

File: services/api/tests/routes/test_croissant.py (hunk 4 above)
Comment (ccl-core) [reply services/api/tests/routes/test_croissant.py:78]: Done.

File: services/api/src/api/routes/croissant.py (hunk 1 above)
Comment (ccl-core) [reply services/api/src/api/routes/croissant.py]: Totally, I forgot that!

File: services/api/src/api/routes/croissant.py [hunk 5]
```diff
@@ -169,50 +186,66 @@ def get_croissant_from_dataset_infos(
...
+                "field": f"{prefix}:field",
+                "fileProperty": f"{prefix}:fileProperty",
+                "fileObject": f"{prefix}:fileObject",
+                "fileSet": f"{prefix}:fileSet",
+                "format": f"{prefix}:format",
+                "includes": f"{prefix}:includes",
+                "isEnumeration": f"{prefix}:isEnumeration",
+                "isLiveDataset": f"{prefix}:isLiveDataset",
+                "jsonPath": f"{prefix}:jsonPath",
+                "key": f"{prefix}:key",
```
Comment (marcenacp) [inline services/api/src/api/routes/croissant.py:230]: ml:key -> sc:key

**Comment by ccl-core** [comment]
Closed until we can reopen it!

File: services/api/src/api/routes/croissant.py [hunk 6]
```diff
@@ -121,39 +125,52 @@ def get_croissant_from_dataset_infos(
...
-                    }
+                    _remove_none_values(
+                        {
+                            "@type": "ml:Field",
+                            "@id": _escape_name(column, fields_names) if isV1 else None,
+                            "name": _escape_name(column, fields_names),
+                            "description": f"Image column '{column}' from the Hugging Face parquet file.",
+                            "dataType": "sc:ImageObject",
+                            "source": {
+                                "fileSet": {"@id": distribution_name},
```
Comment (marcenacp) [inline services/api/src/api/routes/croissant.py:154]: This is also v0
